catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/MetacardAttributeDescriptor.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data.dynamic.impl;
+
+import org.apache.commons.beanutils.DynaProperty;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+
+/**
+ * Implements the {@link AttributeDescriptor} and {@link AttributeType} for dynamic
+ * metacards. Adds the properties that the catalog provider use - indexed, stored, multiValued,
+ * and tokenized.
+ */
+public class MetacardAttributeDescriptor implements AttributeDescriptor, AttributeType {
+    protected MetacardPropertyDescriptor propertyDescriptor;
+
+
+    /**
+     * Creates a MetacardAttributeDescriptor given a {@link DynaProperty} descriptor.
+     * @param descriptor {@link DynaProperty} descriptor for this attribute
+     */
+    public MetacardAttributeDescriptor(DynaProperty descriptor) {
+        if (descriptor instanceof MetacardPropertyDescriptor) {
+            propertyDescriptor = (MetacardPropertyDescriptor) descriptor;
+        } else {
+            propertyDescriptor = new MetacardPropertyDescriptor(descriptor, true, false, true);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return propertyDescriptor.getName();
+    }
+
+    @Override
+    public boolean isMultiValued() {
+        return false;
+    }
+
+    @Override
+    public AttributeType<?> getType() {
+        return this;
+    }
+
+    @Override
+    public boolean isIndexed() {
+        return propertyDescriptor.isIndexedBySource();
+    }
+
+    @Override
+    public boolean isTokenized() {
+        return propertyDescriptor.isTokenized();
+    }
+
+    @Override
+    public boolean isStored() {
+        return false;
```

Comment:
Only in the constructor, similar to the indexed (by the source), tokenized values as well.